Lib/datetime.py

```diff
@@ -692,9 +692,12 @@ def __new__(cls, year, month=None, day=None):
 
         year, month, day (required, base 1)
         """
-        if month is None and isinstance(year, bytes) and len(year) == 4 and \
-                1 <= year[2] <= 12:
+        if (month is None and
+            isinstance(year, (bytes, str)) and len(year) == 4 and
+            1 <= ord(year[2:3])&0xFF <= 12):
```

Comment:
I don't understand how the last condition works.   Is it supposed to allow str year with year[2] outside of 8-bit range?